Python/pystrtod.c

```diff
@@ -918,6 +918,18 @@ char * PyOS_double_to_string(double val,
                       (flags & Py_DTSF_ALT ? "#" : ""), precision,
                       format_code);
         _PyOS_ascii_formatd(buf, bufsize, format, val, precision);
+
+        if (flags & Py_DTSF_NO_NEG_0 && buf[0] == '-') {
```

Comment:
Please could you add some parentheses here? `==` binds more tightly than `&`, so I'm not sure this does the intended thing. (It would be good to find a way to test this code, if only manually.)

Reply:
Sorry, brain fail. This is fine, of course - the relative precedence of `==` and `&` isn't relevant. It might still be worth some parentheses for clarity, though.